pkg/autodiscovery/providers/kube_endpoints.go

```diff
@@ -137,6 +151,34 @@ func (k *KubeEndpointsConfigProvider) invalidateIfChanged(old, obj interface{})
 	}
 }
 
+func (k *KubeEndpointsConfigProvider) invalidateIfChangedEndpoints(old, obj interface{}) {
+	// Cast the updated object, don't invalidate on casting error.
+	// nil pointers are safely handled by the casting logic.
+	castedObj, ok := obj.(*v1.Endpoints)
+	if !ok {
+		log.Errorf("Expected an Endpoints type, got: %v", obj)
```

Comment:
if you want the type you should use: `%T`

```suggestion
		log.Errorf("Expected an Endpoints type, got: %T", obj)
```